lib/Backend/GlobOptBailOut.cpp

```diff
@@ -15,52 +15,258 @@ GlobOpt::CaptureValue(BasicBlock *block, StackSym * stackSym, Value * value, Bai
+
+            // copy unchanged sym to new capturedValues
+            while (sym && sym->m_id < symId)
+            {
+                Assert(sym->IsStackSym());
+                if (!sym->AsStackSym()->HasArgSlotNum() || block->globOptData.capturedArgs->Test(sym->m_id))
+                {
+                    bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, sym->AsStackSym(), iterConst.Data().Value());
+                }
+
+                hasConstValue = iterConst.Next();
+                sym = hasConstValue ? iterConst.Data().Key() : nullptr;
+            }
+            if (sym && sym->m_id == symId)
+            {
+                hasConstValue = iterConst.Next();
+            }
+            if (symId != Js::Constants::InvalidSymID)
+            {
+                // recapture changed constant sym
+
+                symIdBucket = block->globOptData.symToValueMap->GetBucket(symId);
+                if (symIdBucket == nullptr)
+                {
+                    continue;
+                }
+
+                sym = symIdBucket->value;
+                Assert(sym->IsStackSym() && (sym->AsStackSym()->HasByteCodeRegSlot() || sym->AsStackSym()->HasArgSlotNum()));
+
+                val =  block->globOptData.capturedArgs->Test(symId) ? FindValue(sym) : symIdBucket->element;
+                ValueInfo* valueInfo = val->GetValueInfo();
+
+                if (valueInfo->GetSymStore() != nullptr)
+                {
+                    int32 intConstValue;
+                    BailoutConstantValue constValue;
+
+                    if (valueInfo->TryGetIntConstantValue(&intConstValue))
+                    {
+                        constValue.InitIntConstValue(intConstValue);
+                        bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, sym->AsStackSym(), constValue);
+                        symIdHandled = true;
+                    }
+                    else if(valueInfo->IsVarConstant())
+                    {
+                        constValue.InitVarConstValue(valueInfo->AsVarConstant()->VarValue());
+                        bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, sym->AsStackSym(), constValue);
+                        symIdHandled = true;
+                    }
+                }
+                else if (!valueInfo->HasIntConstantValue())
+                {
+                    continue;
+                }
+            }
+
+            if (symIdHandled)
+            {
+                continue;
+            }
+
+            sym = hasCopyPropSym ? iterCopyPropSym.Data().Key() : nullptr;
+
+            while (true)
+            {
+                StackSym * copyPropSym = (sym != nullptr) ? iterCopyPropSym.Data().Value() : nullptr;
+                Sym * mergeSym = nullptr;
+                bool mergeSymId = false;
+
+                // copy unchanged copy prop sym
+                if (sym && !tempBv->Test(copyPropSym->m_id) && sym->m_id < symId)
```

Comment:
Will update the order, my previous thought is put condition with more selectivity (not almost always true) in front.

---
In reply to: [68346499](https://github.com/Microsoft/ChakraCore/pull/1152#discussion_r68346499) [](ancestors = 68346499)